### ISSUE TITLE
:lt() and :gt() selectors

### ISSUE THREAD

**Issue by sm-ghavami** [facelessuser/soupsieve#203]
In the Jsoup we have `:lt(n)` and `:gt(n)` selectors to select elements with index little/greater than n. I think it is nice to have these selectors alternatives in the soupsieve.

**Comment by facelessuser** [comment]
I imagine if we added something like this we'd want to kind of borrow from the latest CSS syntax in regards to `nth-` selectors: `:-soup-lt(3 of <selector>)`.

Something to consider.

@gir-bot add T: feature, C: css-custom
@gir-bot remove S: triage

**Comment by facelessuser** [comment]
The one argument against this is that these selectors are not explicitly needed as they can be covered by existing `:nth-` selectors using `n`.

If there is any reason I would not include these, it is because of this reason.

**Comment by sm-ghavami** [comment]
I doubt that these selectors can be covered by existing selectors. How could you select elements which indexes are greeter than `n` using current selectors?

**Comment by facelessuser** [comment]
```py
from bs4 import _soup

html = """
<div>
<p>One</p>
<p>Two</p>
<p>Three</p>
<p>Four</p>
<p>Five</p>
</div>
"""

soup = _soup(html, features='html5lib')
print('--- Greater than 2 ---')
print(soup.select('p:nth-child(n+3)'))

print('--- Less than 4 ---')
print(soup.select('p:nth-child(-n+3)'))
```

Output

```
--- Greater than 2 ---
[<p>Three</p>, <p>Four</p>, <p>Five</p>]
--- Less than 4 ---
[<p>One</p>, <p>Two</p>, <p>Three</p>]
```

Am I missing something?

**Comment by facelessuser** [comment]
Keep in mind, maybe I don't understand exactly what jsoup does with `lt()` and `gt()`, I'm only basing this off of what I read from the jsoup docs.

**Comment by sm-ghavami** [comment]
You're right.
There are no need for such selectors as they can be covered by `:nth-` selectors.

**Comment by facelessuser** [comment]
Yeah. When I first created soupsieve, I think I looked at some similar selectors before but came to the same conclusion. While I guess the shorthand is "more" convenient, it just didn't seem like it added enough to warrant including them. To be honest, if I implemented it, it would end up being basically an alias for what I showed above, and that is why I think I would probably consider not doing them. It's just quite trivial to do the same thing with `:nth-` selectors.
